x-pack/platform/plugins/shared/fleet/common/services/limited_package.ts

```diff
@@ -17,10 +17,7 @@ export const doesAgentPolicyAlreadyIncludePackage = (
   agentPolicy: AgentPolicy,
   packageName: string
 ): boolean => {
-  if (!agentPolicy.package_policies) {
```

Comment:
I think we may want to keep that one, sometime we retrieve agent policy without `package_policies` so to be sure we pass the correct agent policy. In an ideal world it could be great to have two types one for AgentPolicy one for AgentPolicyWithPackagePolicies to avoid this kind of dynamic check

Reply:
I reverted this change, it was trying to fix an issue I found with switching between agent-based/agentless and new hosts/existing hosts, but I fixed it elsewhere